From: Zhang Qiang <qiang.zhang@windriver.com>
Date: Wed, 27 May 2020 15:57:15 +0800
Subject: workqueue: Remove unnecessary kfree() call in rcu_free_wq()
Git-commit: 342ed2400b78072cc01c0130ce41240dec60d56d
Patch-mainline: 5.8-rc1
References: git-fixes

The data structure member "wq->rescuer" was reset to a null pointer
in one if branch. It was passed to a call of the function "kfree"
in the callback function "rcu_free_wq" (which was eventually executed).
The function "kfree" does not perform more meaningful data processing
for a passed null pointer (besides immediately returning from such a call).
Thus delete this function call which became unnecessary with the referenced
software update.

Fixes: def98c84b6cd ("workqueue: Fix spurious sanity check failures in destroy_workqueue()")

Suggested-by: Markus Elfring <Markus.Elfring@web.de>
Signed-off-by: Zhang Qiang <qiang.zhang@windriver.com>
Signed-off-by: Tejun Heo <tj@kernel.org>
Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 kernel/workqueue.c |    1 -
 1 file changed, 1 deletion(-)

--- a/kernel/workqueue.c
+++ b/kernel/workqueue.c
@@ -3496,7 +3496,6 @@ static void rcu_free_wq(struct rcu_head
 	else
 		free_workqueue_attrs(wq->unbound_attrs);
 
-	kfree(wq->rescuer);
 	kfree(wq);
 }
 
